src/main/java/frc/robot/Robot.java

```diff
@@ -38,6 +41,7 @@ public static enum RobotMode {
   private final SwerveSubsystem swerve =
       new SwerveSubsystem(
           mode == RobotMode.REAL ? new GyroIOPigeon2() : new GyroIO() {},
+          mode == RobotMode.REAL ? new VisionIOReal() : new VisionIOSim(),
```

Comment:
probably should have a way to input multiple visionio instances, maybe make it an array in the constructor. if you do that the modules should also be an array for consistency

Reply:
can i do the thing with the ellipses in the constructor

Reply:
youve prob figured it out by now but you can only do that for the last item in the constructor